contracts/BatchExchangeViewer.sol

```diff
@@ -220,4 +259,10 @@ contract BatchExchangeViewer {
         }
         return updateSellTokenBalance(element, sellTokenBalance);
     }
+
+    function setLength(bytes memory buffer, uint256 length) public pure {
```

Comment:
I think, as this function uses inline assembly, it deserves some doc-comments. Maybe include things like the assumptions being made by the method (like that `buffer.capacity` >= `length`, where `buffer.capacity` is the reserved memory region created by the Solc compiler).